lib/fencing/st_output.c

```diff
@@ -120,19 +120,24 @@ stonith__history_description(stonith_history_t *history, bool full_history,
                        NULL);
 
         // For completed actions, add completion time
-        if (completed_time != NULL) {
+        if (completed_time_s != NULL) {
+            const char *result_s = NULL;
+
             if (full_history) {
-                g_string_append(str, ", completed");
+                result_s = "completed";
             } else if (history->state == st_failed) {
-                g_string_append(str, ", last-failed");
+                result_s = "last-failed";
             } else {
-                g_string_append(str, ", last-successful");
+                result_s = "last-successful";
             }
-            pcmk__g_strcat(str, "='", completed_time, "'", NULL);
+
+            g_string_append_printf(str, ", %s='%s', %s-ns=%ld",
+                                   result_s, completed_time_s,
+                                   result_s, history->completed_nsec);
```

Comment:
We have pcmk__time_hr_t which is currently crm_time_t plus useconds, and useconds should be sufficient for this display here, so let's trim the ns to us and have an equivalent string function for the high-resolution time. That way, we can show the extra detail naturally in the string as a decimal after the seconds.

Reply:
Only for non-XML, right? The XML would go from an int to a float, which I don't think we can do (backward compatibility)

Reply:
Right, just for non-XML. 

pcmk__time_format_hr() might be sufficient already

Reply:
Whatever works. Looking at it again, I think usec would be useful to display in the "friendly" output as well, so at least it will get more use. :)

Reply:
> The XML would go from an int to a float

When I wrote that I was thinking we used the epoch time in the XML. We actually use the string time (date/time/timezone). We probably should still leave the existing alone though...

Reply:
I'm fine with using the same string in all places

Reply:
Works for me and that's simpler. We do have nanoseconds available if we decide we want XML to be higher-res than text though.

Reply:
> nanoseconds available

Never mind, that would be doable but nontrivial in the current framework